From 6a5014e6976cde70d06e5e3c1206abb614d599b3 Mon Sep 17 00:00:00 2001
From: Yong Zhi <yong.zhi@intel.com>
Date: Wed, 28 Jun 2017 11:43:45 -0500
Subject: FROMLIST: videodev2.h add IPU3 meta buffer format

Add the IPU3 specific processing parameter format
V4L2_PIX_FMT_IPU3_PARAMS and metadata formats
for 3A and other statistics:

      V4L2_META_FMT_IPU3_PARAMS
      V4L2_META_FMT_IPU3_STAT_3A
      V4L2_META_FMT_IPU3_STAT_DVS
      V4L2_META_FMT_IPU3_STAT_LACE

Change-Id: Iea772c3af54290e80e15affb7a1a4172bae20c4a
Signed-off-by: Yong Zhi <yong.zhi@intel.com>
(am from https://patchwork.linuxtv.org/patch/41874/)
---
 include/uapi/linux/videodev2.h | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/include/uapi/linux/videodev2.h b/include/uapi/linux/videodev2.h
index ac74031cb37f..911f7fd40fbd 100644
--- a/include/uapi/linux/videodev2.h
+++ b/include/uapi/linux/videodev2.h
@@ -652,6 +652,12 @@ struct v4l2_pix_format {
 /* Flags */
 #define V4L2_PIX_FMT_FLAG_PREMUL_ALPHA	0x00000001
 
+/* Vendor specific - used for IPU3 camera sub-system */
+#define V4L2_META_FMT_IPU3_PARAMS	v4l2_fourcc('i', 'p', '3', 'p') /* IPU3 params */
+#define V4L2_META_FMT_IPU3_STAT_3A	v4l2_fourcc('i', 'p', '3', 's') /* IPU3 3A statistics */
+#define V4L2_META_FMT_IPU3_STAT_DVS	v4l2_fourcc('i', 'p', '3', 'd') /* IPU3 DVS statistics */
+#define V4L2_META_FMT_IPU3_STAT_LACE	v4l2_fourcc('i', 'p', '3', 'l') /* IPU3 LACE statistics */
+
 /*
  *	F O R M A T   E N U M E R A T I O N
  */
-- 
2.7.4

